sapi/fpm/fpm/fpm_sockets.c

```diff
@@ -65,10 +65,18 @@ static void fpm_sockets_cleanup(int which, void *arg) /* {{{ */
 			close(ls->sock);
 		} else { /* on PARENT EXEC we want socket fds to be inherited through environment variable */
 			char fd[32];
+			char *tmpenv_value;
 			sprintf(fd, "%d", ls->sock);
 
 			socket_set_buf = (i % FPM_ENV_SOCKET_SET_SIZE == 0 && i) ? 1 : 0;
-			env_value = realloc(env_value, p + (p ? 1 : 0) + strlen(ls->key) + 1 + strlen(fd) + socket_set_buf + 1);
+			tmpenv_value = realloc(env_value, p + (p ? 1 : 0) + strlen(ls->key) + 1 + strlen(fd) + socket_set_buf + 1);
+			if (!tmpenv_value) {
+				zlog(ZLOG_SYSERROR, "failed to allocate environment");
```

Comment:
It might be good to log a bit more explaining message as it could technically mean that only some socket will be inherited. I thought about that logic quite a bit and think this makes sense. I'm aware that this won't really happen often if ever but a good message can help in understanding of the code.

I will also add a note for myself to try to avoid realloc on every on every set - there's some space for improvements in this code.